chap-cheri-riscv.tex

```diff
@@ -678,103 +678,57 @@ \subsubsection{Non-Compressed Instructions Affected by Capability Encoding
-replace floating-point double operations with capability operations:
-\texttt{C.FLDSP} with \texttt{C.LCSP} and \texttt{C.FSDSP} with
-\texttt{C.SCSP}.
-
-When in capability
-encoding mode, as with load-store instructions, we interpret existing compressed
-instructions \insnnoref{C.J}, \insnnoref{C.JAL}, \insnnoref{C.JR}, and
-\insnnoref{C.JALR} as the capability instructions \insnriscvref{CJAL}
-and \insnriscvref{CJALR}, accepting capability rather than
-integer register operands for jump target registers and link registers.
-
-There is one large gap in the compressed instruction encoding at
-\verb#100X_XXXX_XXXX_XX00# (where \verb#X#$=$ don't care) that could
-be used to support a \insnriscvref{CIncOffsetImm} (\texttt{c.cincoff})
-instruction to allow the stack pointer to be adjusted.
-
-This would result in capability-aware prologue and epilogues:
-
-\begin{small}
-\begin{verbatim}
-   prologue_cap_aware:    epilogue_cap_aware:
-     c.cincoff csp, -16     c.lcsp cra, 8(csp)
-     c.scsp cra, 8(csp)     c.lcsp s0, 0(csp)
-     c.scsp cs0, 0(csp)     c.cincoff csp, 16
-     c.cjr ct0              c.cjr cra
-\end{verbatim}
-\end{small}
-
-A further interaction relates to encoding-mode selection.
-If we pursue a design choice using the lowest bit of a target jump address to
-set the encoding used following a jump, then no additional opcode pressure is
-introduced.
-If we instead choose to use new instructions, such as to get or set the mode
-explicitly, then additional space might be required.
-The amount of space required would be modest, but this additional usage might
-be a further consideration in the encoding-mode management strategy.
-
-If the approach of using a mode bit for compressed instructions is
-adopted, then it follows that the uncompressed versions of the
-instructions should follow suit to preserve the design intent that
-all compressed instructions can be expanded out to uncompressed
-instructions by the processor's decoder, and that the assembler should
-be permitted to optimize uncompressed instructions into their
-compressed form where possible.
-\pdrnote{These last two paragraphs do seem a bit stale, and do we need to
-add a note about the prologue assuming it's been called in the correct
-mode? (or doing a check)}
+For some instructions, the choice to use capability semantics in
+capability encoding mode is straightforward.  Compressed loads and
+stores should use capability-relative addresses just as for
+non-compressed instructions.  Similarly, compressed jump instructions
+should use capability registers for jump target registers and link
+registers.  Finally, instructions which use the stack pointer as an
+implicit operand should use \CSP{} as the stack pointer.  This
+includes loads and stores as well as the stack addressing instructions
+\insnnoref{C.ADD16SP} and \insnnoref{C.ADDI4SPN}.
+
+For other instructions, the decision is less clear.  Pure capability
```

Comment:
I guessed that this was the rationale for other instructions like C.MV and C.ADDI.